### PR TITLE
glib: upgrade requirements

### PR DESCRIPTION
**Pull request by ericLemanissier**
Specify library name and version:  **glib/***

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [X] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by ericLemanissier** [comment]
this is blocked by https://github.com/conan-io/conan-center-index/pull/2862

File: recipes/glib/all/conanfile.py [hunk 1]
```diff
@@ -53,20 +52,22 @@ def config_options(self):
...
             self.requires("libelf/0.8.13")
         if self.settings.os == "Linux":
             if self.options.with_mount:
                 self.requires("libmount/2.33.1")
             if self.options.with_selinux:
-                self.requires("libselinux/2.9")
+                self.requires("libselinux/3.1")
         else:
             # for Linux, gettext is provided by libc
             self.requires("libgettext/0.20.1")
```
Comment (Dark-Passenger) [inline recipes/glib/all/conanfile.py:73]: Can libgettext also be updated ? I believe libgettext has been depreciated for gettext

Comment (ericLemanissier) [reply recipes/glib/all/conanfile.py:73]: not until until https://github.com/conan-io/conan-center-index/pull/1424 is merged

Comment (Dark-Passenger) [reply recipes/glib/all/conanfile.py:73]: my bad saw the depreciation notice didn't know the package was pending.

**Review by uilianries — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Review by danimtb — APPROVED** [review]

**Pull request merged**